src/TMapLabel.h

```diff
@@ -0,0 +1,62 @@
+#ifndef MUDLET_TMAPLABEL_H
+#define MUDLET_TMAPLABEL_H
+
+/***************************************************************************
+ *   Copyright (C) 2008-2013 by Heiko Koehn - KoehnHeiko@googlemail.com    *
+ *   Copyright (C) 2014 by Ahmed Charles - acharles@outlook.com            *
+ *   Copyright (C) 2014-2016, 2018-2020 by Stephen Lyons                   *
+ *                                               - slysven@virginmedia.com *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
+ ***************************************************************************/
+
+
+#include "pre_guard.h"
+#include <QtGlobal>
+#include <QColor>
+#include <QPixmap>
+#include <QSizeF>
+#include <QVector3D>
+#include "post_guard.h"
+
+
+class TMapLabel
+{
+public:
+    TMapLabel()
+    : fgColor(Qt::black)
+    , bgColor(Qt::black)
+    , highlight(false)
+    , showOnTop(false)
+    , noScaling(false)
+    {
+    }
+
+    QVector3D pos;
+    QSizeF size;
+    QSizeF clickSize;
+    QString text;
+    QColor fgColor;
+    QColor bgColor;
+    QPixmap pix;
+    bool highlight;
+    bool showOnTop;
+    bool noScaling;
```

Comment:
Why don't we try this (pseudocode) and remove the duplication in the constructor?

```suggestion
    QColor fgColor {Qt::black};
    QColor bgColor {Qt::black};
    QPixmap pix;
    bool highlight {};
    bool showOnTop {};
    bool noScaling {};
```

Reply:
It makes sense - especially given #4578 - may have been done and gotten lose when I was resolving a merge conflict (where I got the `TMapLabel` class being declared both in it's own header {from my code} and still defined in the `TMap` one {from the main development branch code})...

Reply:
Um, that seems to mean that we end up with a *default* constructor!?

Reply:
?